### DISCUSSION TITLE
Installing testcontainers requires a lot of build tools

### DISCUSSION THREAD

**Discussion by armen-ch** [testcontainers/testcontainers-node#287] [Q&A] [answered]
Hello @cristianrgreco ,

At first, thank you for contributing to the nodejs version of the package. This is a great feature. We are integrating micrioservice component and contract testing in our organization. We are doing this by creating isolated ecosystem for the microservices in the CI/CD pipeline. We use testcontainers to spawn required components (db, cache mem, etc) in docker container. Our docker images install npm packages during docker container build. Installing testcontainer npm requires a lot of build tooling like make, cmake,  gcc, g++. python, etc. This is because testcontainers utilizes cpu-features package as dependency which needs to be built by node-gyp. This in turn makes our base docker images pretty huge something like 2.5G+. Is it possible to avoid package dependency which require build by aforementioned packages/tools? Or may be you can suggest some good solution here.

Thanks

**Comment by cristianrgreco** [comment, marked as answer]
Hi @armen-ch

I've run an `npm list` and I see we're pulling in `cpu-features` as a dependency of `ssh2`.  Looking at the `ssh2` docs:

> cpu-features is set as an optional package dependency (you do not need to install it explicitly/separately from ssh2) that will be automatically built and used if possible. See the project's documentation for its own requirements.
This addon is currently used to help generate an optimal default cipher list

I think then when you run your build, you can do `npm ci --no-optional` to skip it?

**Reply by armen-ch** [reply]
Didn't notice this optional thing. So, it works.
Thank you!
